starlette/routing.py

```diff
@@ -666,9 +679,23 @@ async def lifespan(self, scope: Scope, receive: Receive, send: Send) -> None:
         """
         started = False
         app = scope.get("app")
+        state = scope.get("state")
         await receive()
+        lifespan_needs_state = (
+            len(inspect.signature(self.lifespan_context).parameters) == 2
+        )
+        server_supports_state = state is not None
+        if lifespan_needs_state and not server_supports_state:
+            raise RuntimeError(
+                'The server does not support "state" in the lifespan scope.'
```

Comment:
I removed the "please change server", and use the created the `state` variable to save one line below... 😬